modules/data.remote/R/GEDI_AGB_prep.R

```diff
@@ -0,0 +1,527 @@
+  # if (file.exists(file.path(outdir, "extends.txt"))) {
+  #   pre.extends <- utils::read.table(file.path(outdir, "extends.txt"), skip = 1)
+  #   extends <- rbind(pre.extends, extends) %>% `colnames<-`(c("ymax", "ymin", "xmin", "xmax"))
+  #   extends <- extends[!base::duplicated(extends),]
+  #   utils::write.table(extends, file = file.path(outdir, "extends.txt"), row.names = F)
+  # } else {
+  #   extends <- data.frame(matrix(extends, nrow = 1)) %>% purrr::set_names(c("ymax", "ymin", "xmin", "xmax"))
+  #   utils::write.table(extends, file = file.path(outdir, "extends.txt"), row.names = F)
+  # }
+  # delete downloaded H5 files.
+  unlink(list.files(outdir, "*.h5", full.names = T), recursive = T)
+  if (length(keep.ind) == 0) {
+    return(NA)
+  } else {
+    return(res[keep.ind,])
+  }
+}
+#' DOWNLOAD GEDI level 4A data from DAACL.ORNL
+#'
+#' Download all GEDI footprints from
+#' \href{https://daac.ornl.gov/cgi-bin/dsviewer.pl?ds_id=2056}{the official repository} that intersect a study area, defined as an extent in lon/lat
+#' coordinates. The footprints are located within the global latitude band
+#' observed by the International Space Station (ISS), nominally 51.6 degrees N
+#' and S and reported for the period 2019-04-18 to 2020-09-02
+#'
+#' @param ul_lat Numeric: upper left latitude.
+#' @param lr_lat Numeric: lower right latitude.
+#' @param ul_lon Numeric: upper left longitude.
+#' @param lr_lon Numeric: lower right longitude.
+#' @param ncore Numeric: numbers of core to be used if the maximum core
+#'   available is less than the number of files to be download. Default to the
+#'   number of cores available minus one.
+#' @param from Character: date from which the data search starts. In the form
+#'   "yyyy-mm-dd".
+#' @param to Character: date on which the data search end. In the form
+#'   "yyyy-mm-dd".
+#' @param outdir Character: path of the directory in which to save the
+#'   downloaded files.Default to the working directory. If it doesn't exist it
+#'   will be created. Ignored if \code{just_path=TRUE}
+#' @param just_path Logical: if TRUE return a character vector of available
+#'   files without downloading them. Default to FALSE.
+#' @param subset Numeric vector of indices for downloading a subset of files
+#'   instead of all. If is not numeric it will be ignored silently.
+#' @details During the first use, users will be ask to enter their Earth Explore
+#'   login Information for downloading the data. If you don't have already an
+#'   account, register
+#'   \href{https://urs.earthdata.nasa.gov/users/new?client_id=YQOhivHfMTau88rjbMOVyg&redirect_uri=https%3A%2F%2Fdaac.ornl.gov%2Fcgi-bin%2Furs%2Furs_logon_proc.pl&response_type=code&state=https%3A%2F%2Fdaac.ornl.gov%2Fcgi-bin%2Fdataset_lister.pl%3Fp%3D40}{at
+#'   this link}. These information will be saved in \code{outdir} as a netrc
+#'   file. This function uses the \code{\link[foreach]{foreach}} package for
```

Comment:
1. I moved the creation of the credential file to the `GEDI_AGB_prep` function so that it will be only called once.
2. I documented the usage of the `netrc` file within the `GEDI_AGB_prep` function.